### PR TITLE
Compile jxml files for production use

### PR DESCRIPTION
**Pull request by tgohn**
### WIP

@yangbin @zz85 @jieyanhuang

### CODE REVIEW

**Comment by zz85** [comment]
are `dist/*` files output from gulp? if so, is it better not to be checked in?

**Comment by tgohn** [comment]
yup, `dist/*` files are output from the gulp task.
We need to check them in for usage in browser environment, following https://github.com/systemjs/systemjs

File: tasks/gulp-jxml.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+				cb(null, file);
+			});
+		}
+	);
+}
+
+// callback style
+exports.jxml2System = function jxml2System(filePath, cb) {
+	var builder = require('systemjs-builder');
```
Comment (zz85) [inline tasks/gulp-jxml.js]: requires could be moved to the top

**Comment by zz85** [comment]
i'm not convinced we should check in dist files right now - it feels like its adding more noise to the repo and compiled files might be taken care with @yangbin's new repo server.

also, structure gulp tasks with
```
*
 |- gulp
     |- tasks
     |- utils
```
?

This allows to add a gulpfile which runs gulp/tasks/*

**Comment by yangbin** [comment]
Agree on leaving out the `dist` files. History would need to be rewritten.

**Comment by tgohn** [comment]
- removed `dist/` folder
- about file structure of `tasks/` folder. Let's do a structure changes when we have more files there. Right now, there are only 3 inside.

**Comment by tgohn** [comment]
this project dieded

**Comment by zz85** [comment]
😢

**Comment by yangbin** [comment]
eh not true

**Pull request closed without merging**